core/src/actions_normalize.cpp

```diff
@@ -301,6 +301,17 @@ bool km::core::actions_update_app_context_nfu(
     return false;
   }
 
+  // Strip markers from returned context
+
+  int i, j;
+  for(i = 0, j = 0; items[i].type != KM_CORE_CT_END; i++) {
+    if(items[i].type == KM_CORE_CT_CHAR) {
```

Comment:
Would this be more robust? Since the wording is 'remove markers'

```suggestion
    if(items[i].type != KM_CORE_CT_MARKER) {
```